src/hotspot/cpu/x86/assembler_x86.cpp

```diff
@@ -6305,8 +6393,8 @@ void Assembler::sqrtss(XMMRegister dst, Address src) {
   emit_operand(dst, src, 0);
 }
 
-void Assembler::stmxcsr( Address dst) {
-  if (UseAVX > 0 ) {
+void Assembler::stmxcsr(Address dst) {
+  if (UseAVX > 0 && !UseAPX  ) {
```

Comment:
The !UseAPX test was added because if UseAPX is enabled we want to support extended register use via rex2 encoding in the else clause.  The existing vex encoding remains when UseAPX is not enabled. There is a needs_rex2 check of address registers in the call to ::vex_prefix, asserting if UseAPX not enabled.

Reply:
I added a comment on this to the stmxcsr/ldmxcsr functions.

Reply:
okay